### PR TITLE
RATIS-1081. Change NettyServerStreamRpc to write data to StateMachine.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1081

### CODE REVIEW

File: ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java [hunk 1]
```diff
@@ -33,41 +37,76 @@
...
+  public NettyServerStreamRpc(RaftPeer server, StateMachine stateMachine) {
     this.raftServer = server;
-    setupServer();
+    this.stateMachine = stateMachine;
+    this.channelFuture = buildChannel();
+  }
+
+  private CompletableFuture<DataStream> getDataStreamFuture(ByteBuf buf, AtomicBoolean released) {
+    try {
+      // TODO: read the request from buf
```
Comment (amaliujia) [inline ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java:69]: Nit: may create a Jira to track this TODO.

**Comment by amaliujia** [comment]
LGTM

File: ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java (hunk 1 above)
Comment (szetszwo) [reply ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java:69]: Thanks.  Just have filed RATIS-1085.

File: ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java [hunk 2]
```diff
@@ -72,27 +129,47 @@ public void testDataStream(){
...
+    // TODO RATIS-1085: create a RaftClientRequest and put it in the buffer
+    final ByteBuffer buffer = initBuffer(0, size);
+    return out.streamAsync(buffer);
+  }
+
+  static ByteBuffer initBuffer(int offset, int size) {
+    final ByteBuffer buffer = ByteBuffer.allocateDirect(size);
+    final int length = buffer.capacity();
+    buffer.position(0).limit(length);
+    final StringBuilder b = new StringBuilder();
```
Comment (runzhiwang) [inline ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java:165]: `b` seems useless.

Comment (szetszwo) [reply ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStream.java:165]: Yes, it is useless.

**Comment by runzhiwang** [comment]
LGTM. Just wait CI passed.

**Comment by runzhiwang** [comment]
@szetszwo Thanks the patch, I have merged it. The failed ut looks unrelated to this pr. @amaliujia Thanks for review.

**Comment by szetszwo** [comment]
@runzhiwang and @amaliujia thanks for reviewing this.

**Pull request merged**